test/branch.ts

```diff
@@ -40,6 +40,9 @@ describe('Branch module', async () => {
   };
 
   describe('Get branch head', () => {
+    beforeEach(() => {
+      sinon.restore();
```

Comment:
So generally, I will do a few things different here.  I create a sandbox so that I don't worry about test interference:
```js
const sandbox = sinon.createSandbox();
```

Then, in an `afterEach`, I tend to do my restore on the sandbox:
```js
afterEach(() => sandbox.restore());
```

When you go to create a stub, use the `sandbox` instead of `sinon` at the root to avoid weird leaks and confusion in async tests:
```js
const stub = sandbox.stub(myClass, 'myMethod').resolves('🥳');
```

Reply:
I found this on [stackoverflow](https://stackoverflow.com/a/55251560)
> Since sinon@5.0.0, the sinon object is a default sandbox.
Is a distinct sandbox preferred?